pyo3-benches/benches/bench_dict.rs

```diff
@@ -22,14 +22,14 @@ fn iter_dict(b: &mut Bencher<'_>) {
 fn dict_new(b: &mut Bencher<'_>) {
     Python::with_gil(|py| {
         const LEN: usize = 50_000;
-        b.iter(|| (0..LEN as u64).map(|i| (i, i * 2)).into_py_dict(py));
+        b.iter(|| (0..LEN as u64).map(|i| (i, i * 2)).into_py_dict_bound(py));
```

Comment:
To keep the bechmark comparable, it would probably make sense to use `iter_with_large_drop`:

```suggestion
        b.iter_with_large_drop(|| (0..LEN as u64).map(|i| (i, i * 2)).into_py_dict_bound(py));
```

Motivation: the GIL Ref `&PyDict` will live until the `with_gil` closure ends, so the way the benchmark has changed means that we're now measuring freeing the dict when we weren't before.

Reply:
Makes sense 👍